core/mchat.php

```diff
@@ -1056,7 +1056,13 @@ public function assign_messages($rows, $page = '')
 
 			$message_age = time() - $row['message_time'];
 			$minutes_ago = $this->get_minutes_ago($message_age);
-			$datetime = $this->user->format_date($row['message_time'], $this->settings->cfg('mchat_date'), true);
+			// If relative time is selected, also display "today" / "yesterday", else display absolute time.
+			if ($this->settings->cfg('mchat_relative_time'))
+			{
+				$datetime = $this->user->format_date($row['message_time'], $this->settings->cfg('mchat_date'), false);
```

Comment:
Your right, in addition I just encountered another drawback of this method : when the message is posted the relative time function of mchat is working good, but if you do not refresh the page during more than one hour the message is displayed as posted "less than a minute ago" because the title property in the span being copied as is.

Need to find a more nice method to do this (perahps by extanding a bit further the original approach of mchat, calculating Today / Yesterday from the date in JS).

Forget about this merge for now.